consensus/src/consensus_observer/observer/payload_store.rs

```diff
@@ -26,12 +26,12 @@ pub enum BlockPayloadStatus {
 }
 
 /// A simple struct to store the block payloads of ordered and committed blocks
-#[derive(Clone)]
 pub struct BlockPayloadStore {
     // The configuration of the consensus observer
     consensus_observer_config: ConsensusObserverConfig,
 
-    // Block transaction payloads (indexed by epoch and round)
+    // Block transaction payloads (indexed by epoch and round).
+    // This is directly accessed by the payload manager.
```

Comment:
it looks like we already wrap the BlockPayloadStore with Arc of Mutex, we don't need another layer of it here?

Reply:
Yeah, we could pass the larger object to the payload manager now that there's a wider lock. My only concern is if BlockPayloadStore becomes more complex, and we don't really want the payload manager needing the global lock. But, let me see -- I can take a crack at cleaning it up in another PR. 😄